### ISSUE TITLE
Improve check_snmp string handling (version 2.3.3 and current Git code)

### ISSUE THREAD

**Issue by maggu** [nagios-plugins/nagios-plugins#620]
The thing I find most lacking among all Nagios plugins I've used is that check_snmp is pretty bad at handling string data. Here is an excellent example of a widely used case that would be great if check_snmp handled well:

https://exchange.nagios.org/directory/Plugins/Hardware/UPS/Tripp-Lite/check_ups/details

```
$ check_ups.pl -H 2001:db8::1:2:3:4
OK - UPS State is normal(3). Battery State is batteryNormal(2).
```

What it does is check two OIDs (upsOutputSource and upsBatteryStatus), to see that their state is normal. Pretty straightforward.

So let's try to do the same thing with check_snmp instead (with UPS-MIB from [RFC 1628](https://tools.ietf.org/html/rfc1628) installed):

```
$ check_snmp -H 2001:db8::1:2:3:4 -m all -l "UPS State:" -o UPS-MIB::upsOutputSource.0 -l "Battery State:" -o UPS-MIB::upsBatteryStatus.0 -R normal
SNMP OK - UPS State: normal(3) Battery State: batteryNormal(2) |
```

Awesome! The output there looks pretty good. There are just two problems:

1. upsBatteryStatus isn't checked.
2. Everything is OK or CRITICAL, and I can't get a WARNING state.

The only way currently (if I haven't completely missed something) to do those two things is to use thresholds for numerical values:

```
$ check_snmp -H 2001:db8::1:2:3:4 -m all -l "UPS State:" -o UPS-MIB::upsOutputSource.0 -l "Battery State:" -o UPS-MIB::upsBatteryStatus.0 -w '3:3,2:2' -c '@4:5'
SNMP OK - UPS State: 3 Battery State: 2 | 'UPS State:'=3;3:3;@4:5 'Battery State:'=2;2:2;
```

OK. Well, that... kind of works. For upsOutputSource check_ups.pl returns OK if 3, WARNING if 6 or 7, and CRITICAL if 0, 1, 4 or 5. So it's not the same, but maybe good enough. It's just that:

1. The MIB translation of values is lost.
2. It's a pretty ugly and unintuitive solution. Thresholds weren't meant for this.
3. I can't use it if I want to handle actual STRING values.

**So to sum it up:** check_snmp is really awesome if I want to compare numerical values to thresholds. It's not so great for strings.

What I would like to be able to do is something like this (which would be exactly what check_ups.pl does):

```
check_snmp -H 2001:db8::1:2:3:4 -m all -l "UPS State:" -o UPS-MIB::upsOutputSource.0 -W normal -R "normal|booster|reducer" -l "Battery State:" -o UPS-MIB::upsBatteryStatus.0 -W normal
```

If upsOutputSource is not normal(3), booster(6) or reducer(7) I get a CRITICAL state.
If any value (upsOutputSource or upsBatteryStatus) is not normal I get a WARNING state.
Otherwise the state is OK.

In other words I'd like to be able to:

1. Handle multiple OIDs well.
2. Get a WARNING state if an expression doesn't match.
